File: backend/ResQueue/ResQueue/Dtos/UpsertMessageDto.cs
using System.Text.Json.Nodes;

namespace ResQueue.Dtos;

public class UpsertMessageDto
{
    public string BrokerId { get; set; } = null!;
    public string QueueId { get; set; } = null!;
    public JsonNode Body { get; set; } = null!;

    /// <summary>
    ///  "json" or "base64"
    /// </summary>
    public string BodyEncoding { get; set; } = null!;

    public RabbitmqUpsertMessageMetadataDto? RabbitmqMetadata { get; set; }
}

File: frontend/src/dtos/message/upsertMessageDto.ts
import type { RabbitMQUpsertMessageMetaDto } from './rabbitMQUpsertMessageMetaDto'

export interface UpsertMessageDto {
  brokerId: string
  queueId: string
  body: any
  bodyEncoding: 'json' | 'base64' | 'string'
  rabbitmqMetadata?: RabbitMQUpsertMessageMetaDto
}
